### PR TITLE
Support rocky linux in azure cloud provider 

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Rocky linux support in azure
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Support rockylinux in azure 
```

### CODE REVIEW

File: pkg/userdata/ubuntu/testdata/kubelet-version-without-v-prefix.yaml [hunk 1]
```diff
@@ -191,6 +188,16 @@ write_files:
...
 
+- path: "/opt/disable-swap.sh"
+  permissions: "0755"
+  content: |
+    # Make sure we always disable swap - Otherwise the kubelet won't start as for some cloud
+    # providers swap gets enabled on reboot or after the setup script has finished executing.
+    cp /etc/fstab /etc/fstab.orig
+    cat /etc/fstab.orig | awk '$3 ~ /^swap$/ && $1 !~ /^#/ {$0="# commented out by cloudinit\n#"$0} 1' > /etc/fstab.noswap
+    mv /etc/fstab.noswap /etc/fstab
+    swapoff -a
```
Comment (xmudrii) [inline pkg/userdata/ubuntu/testdata/kubelet-version-without-v-prefix.yaml:199]: Any specific reason why we changed this only for Ubuntu? Previously, it was only:

```
sed -i.orig '/.*swap.*/d' /etc/fstab
swapoff -a
```

Comment (moadqassem) [reply pkg/userdata/ubuntu/testdata/kubelet-version-without-v-prefix.yaml:199]: That's actually wrong. This is a copy and paste mistake, will put it back to what it was.

**Review by xmudrii — APPROVED** [review]

**Pull request merged**